### ISSUE TITLE
Decide on new tech stack

### ISSUE THREAD

**Issue by jankoegel** [getAlby/website#169]
# What / Why
+ we're in the process of redesigning the landing pages, which opens the possibility to change the tech stack with which they are built.
+ we have doubts that the current stack (Remix) is the best fit since we don't use it anywhere else in Alby and this repo only serves 3 pages so far.

# Step 1: What are our Requirements
+ what is the purpose of the landing pages and who will use them?

+ who will edit the content of landing pages?
  + [x] developers
  + [x] non-developers who know how to use Git(hub)
  + [ ] non-developers who **don't know** how to use Git(hub)
  + [ ] other (please specify)
+ are there any new features (e.g. complex JS interactions or service integrations) we want to add to landing pages within the next three months beyond what we currently have online? If so, please list them here:
  + ... _add your feature here_
+ is it important that the `website` repo remains open source?
  + [ ] yes
  + [x] no
+ should the `website` repo remain independent or could we just integrate the landing pages in Rails? Maintenance in Rails would be easier but could make open-source contributions rarer? How important are open-source code contributions for our landing pages or won't we just build them ourselves and only accept design inspirations from contributors?
  + [ ] should remain independent
  + [x] can be integrated in Rails (which also means closed source)
+ are there any frontend libraries the `website` repo should use?
  + Tailwind since the extension and Rails app are built with it as well.
+ are there styles / UI elements such as buttons that the landing pages share with other parts of Alby, such as the extension or the Rails app? How important is it that these elements are synced / look the same?
+ would it be critical to know on landing pages whether a user is logged into the Rails app, e.g. to show them a link to their dashboard instead of a login button?
  + [ ] yes, important.
  + [x] no, unimportant.

# Step 2: Potential Tech Stacks
```diff
- We should really only discuss this once the requirements above have been discussed and finalized!
```

### Pros
+ the Rails app is built with the same stack
+ we don't need to maintain a second app if we integrate this with the Rails app
+ when integrated in the Rails app we'll know whether a user is logged in on landing pages and can offer a richer experience
+ very robust, will be maintained for many years to come.
+ simple SSR with JS trickled on top.

### Cons
+ if we don't integrate it with the Rails app Rails is overblown / too complex to just render a bunch of static pages
+ only Bumi and Jan have experience with Rails so far.

### Pros
+ Ruby-based like the Rails app
+ made for static content pages
+ simple SSR

### Cons
+ only Bumi and Jan know Ruby so far

### Pros
+ JS-based like the extension, Saturn 2 might be built with this.
+ designed to generate static websites.
+ open source contributor-friendly

### Cons
+ JS ecosystem is brittle
+ next.js might no longer be maintained in a few years / get replaced by the next hype.
+ DOM hydration / SSR/CSR is complex.

## Remix
+ https://remix.run/blog/remix-vs-next
### Pros
+ ???
### Cons
+ not used anywhere else in Alby so far, which means we need to learn Remix and how to maintain it just for this small repo.

### Questions
+ is Remix intended / a good fit for very simple static pages with little interactivity?

**Comment by jankoegel** [comment]
current prototype with nextJS:
https://github.com/getAlby/website/tree/all-new-2022-10
https://getalby-dev.vercel.app/#

**Comment by jakubszwedo** [comment]
> are there styles / UI elements such as buttons that the landing pages share with other parts of Alby, such as the extension or the Rails app? How important is it that these elements are synced / look the same?

My 2 sats here, as I mentioned on slack:

I do not think it's necessary to use exactly the same UI style across landing/marketing and apps (extension/rails). It is common (at least in web3 design) to have treat apps as different entities from landings/marketings.

You can see examples here:

https://compound.finance/ vs https://app.compound.finance/
https://uniswap.org/ vs https://app.uniswap.org/
https://aave.com/ vs https://app.aave.com/

I took that as a default when inherited current website redesign. I think it's good UX to slightly separate those, especially if we showcase many different products/services on landing page. IMO what is important to unify our "apps" design, so rails and extension, but landing may be a bit more fanciful.

That being said If we still decide to have it all unified, I can work on this but that means some bigger styling changes to either (or rather both) landing page and both apps.

**Comment by jankoegel** [comment]
+ I discussed this with @reneaaron and @bumi yesterday and we filled out the requirements above together.
+ given the requirements and our current tech landscape, we think it makes the most sense to build the new landing pages in Ruby on Rails and integrate them in the existing Rails app.

+ assumptions:
  + people who want to edit content on landing pages know HTML and Github
  + landing pages in Rails are simple enough for the entire team to edit. There will be plenty of existing code that can be copied.
  + Rails app is already running and maintained and offers a very solid and future-proof stack to build on.
  + we should keep things simple and not overthink this since we only have 3 content pages so far :D
